test/API/H5_api_link_test.c

```diff
@@ -846,15 +858,6 @@ test_create_hard_link_invalid_params(void)
         {
             TESTING_2("H5Lcreate_hard across files");
 
-            HDsnprintf(ext_link_filename, H5_API_TEST_FILENAME_MAX_LENGTH, "%s",
```

Comment:
I think this is fine for these cases. We'll probably eventually want some better organization around sub-tests in the future since the idea of creating and deleting the file for each sub-test is to start with a clean state and not have any test inter-dependencies at all; too many times I've encountered problems like that. The whole `PART_BEGIN`/`PART_END` system was a quick hack I wrote to try and keep related sub-tests together, but there are probably better approaches.